src/util/error.c

```diff
@@ -31,12 +31,11 @@ int error_set(error *e, int code, const char *msg) {
 
 int error_fmt(error *e, int code, const char *fmt, ...) {
   va_list ap;
-
   e->code = code;
+#ifdef DEBUG
   va_start(ap, fmt);
   vsnprintf(e->msg, ERROR_MSGBUF_LEN, fmt, ap);
   va_end(ap);
-#ifdef DEBUG
```

Comment:
Is it ever being used w/o DEBUG? The reasons I did it is because I have noticed all tests print humungous number of those messages in DEBUG (without failing - the errors are being ignored) - meaning you are currently spend a lot of CPU time in sprintf() on non-error path.